### PR TITLE
runtimevar/doc: refresh provider implementation godoc

### PR DESCRIPTION
**Pull request by vangent**
Updates #896.

### CODE REVIEW

File: runtimevar/constantvar/constantvar.go [hunk 1]
```diff
@@ -12,8 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package constantvar provides a runtimevar.Driver implementation for variables
-// that never change.
+// Package constantvar provides a runtimevar implementation with variables
+// that never change. Use New, NewBytes, or NewError to construct a
+// runtimevar.Variable.
```
Comment (zombiezen) [inline runtimevar/constantvar/constantvar.go]: Here and elsewhere:

```suggestion
// *runtimevar.Variable.
```

File: runtimevar/runtimeconfigurator/runtimeconfigurator.go [hunk 2]
```diff
@@ -88,7 +98,7 @@ func newWatcher(client pb.RuntimeConfigManagerClient, name ResourceName, decoder
 	}, nil
 }
 
-// ResourceName identifies the full configuration variable path used by the service.
+// ResourceName identifies a full configuration variable path.
```
Comment (zombiezen) [inline runtimevar/runtimeconfigurator/runtimeconfigurator.go]:
```suggestion
// ResourceName identifies a configuration variable.
```

File: runtimevar/filevar/filevar.go [hunk 3]
```diff
@@ -12,22 +12,19 @@
...
 //
 // Known Issues:
 //
-// * On Mac OSX, if user copies an empty file into a configuration file, Watch will not be able to
-// detect the change since event.Op is Chmod only.
-//
-// * Saving a configuration file in vim using :w will incur events Rename and Create. When the
-// Rename event occurs, the file is temporarily removed and hence Watch will return error.  A
-// follow-up Watch call will then detect the Create event.
+// * On Mac OSX, if an empty file is copied into a configuration file,
```
Comment (zombiezen) [inline runtimevar/filevar/filevar.go]:
```suggestion
// * On macOS, if an empty file is copied into a configuration file,
```

File: runtimevar/etcdvar/etcdvar.go [hunk 4]
```diff
@@ -38,11 +38,12 @@ import (
...
 type Options struct{}
 
-// New constructs a runtimevar.Variable object that uses client to watch
-// variables in etcd.
-// Provide a decoder to unmarshal updated configurations into similar
-// objects during the Watch call.
-func New(name string, cli *clientv3.Client, decoder *runtimevar.Decoder, _ *Options) (*runtimevar.Variable, error) {
+// New constructs a runtimevar.Variable that uses client to watch the variable
+// name on an etcd server.
+// Etcd returns raw bytes; provide a decoder to decode the raw bytes into the
```
Comment (zombiezen) [inline runtimevar/etcdvar/etcdvar.go]: I think etcd is always lowercased.

```suggestion
// etcd returns raw bytes; provide a decoder to decode the raw bytes into the
```

File: runtimevar/constantvar/constantvar.go (hunk 1 above)
Comment (vangent) [reply runtimevar/constantvar/constantvar.go]: Done.

**Pull request merged**